File: src/my_py_pkg/my_py_pkg/number_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import Int64
from example_interfaces.srv import SetBool

"""
A ROS2 Node that publishes a number to a topic every 1 second.
"""
class NumberPublisherNode(Node):
    """
    A function that initializes the number publisher and timer for publishing a number
    to the 'number' topic
    """
    def __init__(self):
        super().__init__("number_publisher")
        self.publisher_ = self.create_publisher(Int64, "number", 10)
        self.timer_ = self.create_timer(1, self.publish_number)
        self.get_logger().info("Number publisher has been started.")
    
    """
    A function that publishes an Int64 number to the 'number' topic
    """
    def publish_number(self):
        msg = Int64()
        msg.data = 2
        self.publisher_.publish(msg)


"""
The main function responsible for continuously spinning the NumberPublisherNode.
"""
def main(args=None):
    rclpy.init(args=args)
    node = NumberPublisherNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/number_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int64.hpp"

using namespace std::placeholders;
using namespace std::chrono_literals;

/**
 * A node that creates a publisher for the 'number' topic a number
 * consistently.
 */
class NumberPublisherNode : public rclcpp::Node
{
public:
    NumberPublisherNode() : Node("number_publisher")
    {
        number_publisher_ = this->create_publisher<example_interfaces::msg::Int64>("number", 10);
        timer_ = this->create_wall_timer(0.5s, std::bind(&NumberPublisherNode::publishNumbers, this));

    }
private:
    /**
     * A callback function that prints a number to the 'number' topic.
     */
    void publishNumbers()
    {
        auto msgNumber = example_interfaces::msg::Int64();
        msgNumber.data = 2;
        number_publisher_->publish(msgNumber);

    }
    
    // node instance variables
    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr number_publisher_;
    rclcpp::TimerBase::SharedPtr timer_;

};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<NumberPublisherNode>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}